### PR TITLE
Update workflows to use gradle/actions/setup-gradle (#58628)

### LINKED ISSUE #58628: Switch out the `gradle/gradle-build-action` in favor of `gradle/actions/setup-gradle`
## What problem does this address?
The `gradle/gradle-build-action` has been updated to [version 3.0.0](https://github.com/gradle/gradle-build-action/releases/tag/v3.0.0). The main change in this version is the action will now ask as a pass through to the preferred `gradle/actions/setup-gradle` action.

## What is your proposed solution?
Any workflows using the former should be changed. This should be as simple as changing the `uses` line in our workflows.

### PR DESCRIPTION
**Pull request by shreyash3087**
## What?
Updates workflows to use gradle/actions/setup-gradle instead of gradle/gradle-build-action as the latter action has been superseded in version 3.0.0.

fixes #58628 

## Why?
The `gradle/gradle-build-action` has been updated to version 3.0.0, where it now acts as a pass-through to the preferred `gradle/actions/setup-gradle` action. This change ensures that our workflows stay up-to-date with the latest recommended practices and dependencies.

## How?
The PR modifies the relevant workflows (rnmobile-android-runner.yml) to replace the usage of `gradle/gradle-build-action` with `gradle/actions/setup-gradle`. 

## Testing Instructions

1. Run the modified workflows locally to ensure they execute without errors.
2. Test the workflows in different environments to ensure compatibility.

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 
N/A

### CODE REVIEW

**Comment by shreyash3087** [comment]
@t-hamano I noticed that the previous pull request (#58645) on the issue (#58628) didn't provide much detail about the changes being made, so I went ahead and created a new pull request (#58713) with the same changes but with additional documentation and clarity.

Even though it was a small change,I believe it's important for contributors to have clear information about the purpose and impact of the changes, so I've included detailed explanations in my pull request description. If there are any questions or suggestions, please do let me know.

**Comment by t-hamano** [comment]
Thanks for the PR!

As you say, explaining the what, why, and how in PR is extremely important. However, since I am not familiar with Gutenberg mobile, I will add the mobile team as a reviewer 🙇

File: .github/workflows/rnmobile-android-runner.yml
Comment (geriux) [inline .github/workflows/rnmobile-android-runner.yml]: This is currently failing due to the commit hash I believe 🤔 since it references the `gradle/gradle-build-action` [repo](https://github.com/gradle/gradle-build-action/commit/3b1b3b9a2104c2b47fbae53f3938079c00c9bb87).

I think we should change it to [ec92e829475ac0c2315ea8f9eced72db85bb337a](https://github.com/gradle/actions/commit/ec92e829475ac0c2315ea8f9eced72db85bb337a) which is the `v3.0.0` release in the `gradle/actions` repo.

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

I checked the [the latest run of React Native E2E Tests (Android)](https://github.com/WordPress/gutenberg/actions/runs/7812076264/job/21308371928?pr=58713#step:7:1) and confirmed it worked as expected.

Thanks @shreyash3087 for updating this 🙇 .

**Comment by shreyash3087** [comment]
@fluiddot Thanks for the review and thanks to @geriux for pointing the error out,i was confused about the failing test. But all checks now pass successfully.

**Comment by fluiddot** [comment]
@shreyash3087 before proceeding to merge the PR, let us know if you took a look at [this comment](https://github.com/WordPress/gutenberg/pull/58713#issuecomment-1928880952) related to ensuring your work is properly credited in WordPress releases, and in case you'd like first to link your GitHub and WordPress.org account.

**Comment by shreyash3087** [comment]
@fluiddot Yes i did read that comment.If we talk about linking my github account to wordpress.org,i did try to do that,i followed the link to login to WordPress.org, but it said that it was an unknown email address.I then tried to go ahead and register my account,but then it said "That username is already in use.Your account is pending approval. You will receive an email at Shreyash3087@gmail.com to set your password when approved." It has now been 3 days since i have been trying to link my accounts,but i have not yet received any approval on my mail. Now coming to the credits part i read the Contribution Attribution Page about credits,but since i am beginner in open-source,its just my second contribution and im learning the stuff, i was confused where to put these credits.It says to put it at the bottom of the merge commit message,so is it talking about the description of the commit message? I was also unsure when do i need to put these credits,after approval or before.I just didn't want to mess anything up ,so i ignored the message.

**Comment by fluiddot** [comment]
> If we talk about linking my github account to wordpress.org,i did try to do that,i followed the link to login but it said that it was an unknown email address.I then tried to go ahead and register my account,but then it said "That username is already in use.Your account is pending approval. You will receive an email at [Shreyash3087@gmail.com](mailto:Shreyash3087@gmail.com) to set your password when approved." It has now been 3 days since i have been trying to link my accounts,but i have not yet received any approval on my mail.

Sad to hear you had trouble logging in/registering the account. I'd recommend reaching support at https://wordpress.org/support/forums/.

> Now coming to the credits part i read the Contribution Attribution Page about credits,but since i am beginner in open-source,its just my second contribution and im learning the stuff, i was confused where to put these credits.It says to put it at the bottom of the merge commit message,so is it talking about the description of the commit message? I was also unsure when do i need to put these credits,after approval or before.I just didn't want to mess anything up ,so i ignored the message.

No worries. The credits will be incorporated in the merge commit once we merge the PR, no need to change anything from your side. The only thing to consider is whether you'd like first to link your WordPress.org account to reference it in the merge commit message or just your GitHub account.

**Comment by shreyash3087** [comment]
@fluiddot Thank you for the clarification.I Understand Now. Since I am unsure how long is it gonna take for my Wordpress.org login issue to get resolved,and i can't link my github to WordPress.org without logging in,so I would prefer to have the credits for my contributions solely on GitHub.

**Comment by fluiddot** [comment]
> @fluiddot Thank you for the clarification.I Understand Now. Since I am unsure how long is it gonna take for my Wordpress.org login issue to get resolved,and i can't link my github to WordPress.org without logging in,so I would prefer to have the credits for my contributions solely on GitHub.

Great, I've just gone ahead and merged the PR. Thanks @shreyash3087 for your contribution 🎊 .

**Pull request merged**
